### PR TITLE
Revise - WebEngine support for SDL JavaScript

### PR DESCRIPTION
**Pull request by kshala-ford**
This PR is a revision to #767 WebEngine support for SDL JavaScript including changes as requested in this comment https://github.com/smartdevicelink/sdl_evolution/issues/767#issuecomment-505180444

### CODE REVIEW

**Comment by kshala-ford** [comment]
@theresalech The PR is ready for review.

File: proposals/0240-sdl-js-pwa.md [hunk 1]
```diff
@@ -89,6 +93,22 @@ Due to a new app platform, the hybrid app preference should be modified to track
...
 App registration on the SDL Developer Portal (smartdevicelink.com) should allow a developer to specify an app as a local web app. As the app platforms increase (2 -> 3) a new way to specify preferences should be introduced. The element `BOTH` should be deprecated and replaced by `ALL` for a next major release. Occurrences of `BOTH` would be treated as `ALL`.
 
+### 1.4 HMI API using App Properties RPCs
+
+The HMI API should be extended to set app properties to SDL Core. This addition should allow the HMI to add app policy IDs to the policy database as they are installed by the embedded OEM catalog.
+
+The HMI API extension is mostly a copy of the cloud-app-properties included in the mobile API. 
+
+1. The OEM store uses `SetAppProperties` using the manifest data.
+2. The OEM store may choose add not-installed apps to SDL using `enabled` parameter set to `false` (optional)
```
Comment (theresalech) [inline proposals/0240-sdl-js-pwa.md]:
```suggestion
2. The OEM store may choose to add not-installed apps to SDL using `enabled` parameter set to `false` (optional)
```

File: proposals/0240-sdl-js-pwa.md [hunk 2]
```diff
@@ -123,9 +143,29 @@ Widgets are still available and can be controlled using `Show` RPC. Any overlay
...
+-	NHTSA related guidelines
+  o	Amount of text (button and text fields)
+  o	number of lines of text
+  o	Complexity of use cases (number of steps to fulfill a use case)
+
+More items may be included in the ruleset as they become Driver Distraction affected.
+
+##### 2.2.4 System context and event change
+
+Independent of the app presentation type, the HMI will continue provide system context information from the app. An application which uses the open HMI should continue to recieve `OnHMIStatus` notifications and SDL Core will still be notified about event changes.
```
Comment (theresalech) [inline proposals/0240-sdl-js-pwa.md]:
```suggestion
Independent of the app presentation type, the HMI will continue to provide system context information from the app. An application which uses the open HMI should continue to recieve `OnHMIStatus` notifications and SDL Core will still be notified about event changes.
```

File: proposals/0240-sdl-js-pwa.md [hunk 3]
```diff
@@ -168,48 +208,143 @@ The upside of apps running with a WebEngine is that it comes with an extremely f
...
-  4. If Core doesn't know the app ID it should ask HMI for a policy update.
-     1. Alternatively HMI can call `UpdateSDL` to enforce SDL to peform the policy update.
-2. Policy HMI impl. (incl. SYNCP) should have the ability to use embedded modem to send policy snapshot to the policy server.
-3. All CloudAppProperties RPCs and related structs should also be available through HMI_API
-   1. This allows OEM stores easier communication with SDL.
-4. A new enum called "CoreTransportRole" with "CLIENT" and "SERVER" elements should be added
-   1. CloudAppProperties should have a parameter for "coreTransportRole"
+Local Node.js or Java were considered as alternative options for locally running applications. However both options have downsides:
+1. Both are difficult to sandbox. Compared to a web engine the effort to sandbox a Node.js or java application and to protect the vehicle system are very high.
+2. Limited app availability. App developers would potentially need to reassemble a godd portion of existing code and write new code to make locally running applications possible.
```
Comment (theresalech) [inline proposals/0240-sdl-js-pwa.md]:
```suggestion
2. Limited app availability. App developers would potentially need to reassemble a good portion of existing code and write new code to make locally running applications possible.
```

File: proposals/0240-sdl-js-pwa.md [hunk 4]
```diff
@@ -168,48 +208,143 @@ The upside of apps running with a WebEngine is that it comes with an extremely f
...
+3. Licensing and compatibility. License cost may apply for embedded in-vehicle use. Efforts avoiding license using older versions may cause compatiblity issues leading to code rewrite. Open source variants may cause other license issues.
 
-### Launching an app (alternative solution)
+Many services are available over a web appliation and modern WebEngines provide a good sandboxing ability. Last the ability of an open HMI is outstanding and not supported by either of the alternatives.
```
Comment (theresalech) [inline proposals/0240-sdl-js-pwa.md]:
```suggestion
Many services are available over a web application and modern WebEngines provide a good sandboxing ability. Lastly, the ability of an open HMI is outstanding and not supported by either of the alternatives.
```

**Comment by kshala-ford** [comment]
@theresalech thank you so much for the quick review. I committed all the suggestions. Please let me know if there are other changes to make.

**Pull request merged**